File: app/src/main/java/com/nytimes/android/external/playbillingtester/APIOverrides.java
package com.nytimes.android.external.playbillingtester;

import android.content.SharedPreferences;
import android.support.annotation.NonNull;

import javax.inject.Inject;

/**
 * Wrapper for user preferences such as API overrides and purchases
 */

public class APIOverrides {
    public static final String PREF_NAME = "PlayInApBillingModel";
    public static final String CONFIG_FILE = "playbillingtester.json";
    public static final int RESULT_DEFAULT = -1;            // - no user override
    private static final String IS_BILLING_SUPPORTED = "isBillingSupported";
    private static final String GET_BUY_INTENT = "getBuyIntent";
    private static final String BUY = "Buy";
    private static final String GET_PURCHASES = "getPurchases";
    private static final String GET_SKU_DETAILS = "getSkuDetails";
    private static final String CONSUME_PURCHASE = "consumePurchase";
    private static final String GET_BUY_INTENT_TO_REPLACE_SKUS = "getBuyIntentToReplaceSkus";
    private static final String REPLACE = "Replace";
    private static final String USERS = "Users";
    private static final String DEFAULT_USER = "200nyttest1@nytimes.com";

    @NonNull
    private final SharedPreferences sharedPreferences;

    @Inject
    public APIOverrides(@NonNull SharedPreferences sharedPreferences) {
        this.sharedPreferences = sharedPreferences;
    }

    int getIsBillingSupportedResponse() {
        return sharedPreferences.getInt(IS_BILLING_SUPPORTED, RESULT_DEFAULT);
    }

    void setIsBillingSupportedResponse(int response) {
        sharedPreferences.edit().putInt(IS_BILLING_SUPPORTED, response).apply();
    }

    public int getGetBuyIntentResponse() {
        return sharedPreferences.getInt(GET_BUY_INTENT, RESULT_DEFAULT);
    }

    void setGetBuyIntentResponse(int response) {
        sharedPreferences.edit().putInt(GET_BUY_INTENT, response).apply();
    }
    int getBuyResponse() {
        return sharedPreferences.getInt(BUY, RESULT_DEFAULT);
    }

    void setBuyResponse(int response) {
        sharedPreferences.edit().putInt(BUY, response).apply();
    }

    public int getGetPurchasesResponse() {
        return sharedPreferences.getInt(GET_PURCHASES, RESULT_DEFAULT);
    }

    void setGetPurchasesResponse(int response) {
        sharedPreferences.edit().putInt(GET_PURCHASES, response).apply();
    }
    public int getGetSkuDetailsResponse() {
        return sharedPreferences.getInt(GET_SKU_DETAILS, RESULT_DEFAULT);
    }

    void setGetSkuDetailsResponse(int response) {
        sharedPreferences.edit().putInt(GET_SKU_DETAILS, response).apply();
    }

    public int getConsumePurchaseResponse() {
        return sharedPreferences.getInt(CONSUME_PURCHASE, RESULT_DEFAULT);
    }

    void setConsumePurchaseResponse(int response) {
        sharedPreferences.edit().putInt(CONSUME_PURCHASE, response).apply();
    }

    public int getGetBuyIntentToReplaceSkusResponse() {
        return sharedPreferences.getInt(GET_BUY_INTENT_TO_REPLACE_SKUS, RESULT_DEFAULT);
    }

    void setGetBuyIntentToReplaceSkusResponse(int response) {
        sharedPreferences.edit().putInt(GET_BUY_INTENT_TO_REPLACE_SKUS, response).apply();
    }

    public int getReplaceResponse() {
        return sharedPreferences.getInt(REPLACE, RESULT_DEFAULT);
    }

    void setReplaceReponse(int response) {
        sharedPreferences.edit().putInt(REPLACE, response).apply();
    }

    @NonNull
    String getUsersResponse() {
        return sharedPreferences.getString(USERS, DEFAULT_USER);
    }

    void setUsersReponse(@NonNull String user) {
        sharedPreferences.edit().putString(USERS, user).apply();
    }
}

File: app/src/main/java/com/nytimes/android/external/register/APIOverrides.kt
package com.nytimes.android.external.register

import android.content.SharedPreferences

import javax.inject.Inject

/**
 * Wrapper for user preferences such as API overrides and purchases
 */

class APIOverrides @Inject constructor(private val sharedPreferences: SharedPreferences) {

    var isBillingSupportedResponse: Int
        get() = sharedPreferences.getInt(IS_BILLING_SUPPORTED, RESULT_DEFAULT)
        set(response) = sharedPreferences.edit().putInt(IS_BILLING_SUPPORTED, response).apply()

    var getBuyIntentResponse: Int
        get() = sharedPreferences.getInt(GET_BUY_INTENT, RESULT_DEFAULT)
        internal set(response) = sharedPreferences.edit().putInt(GET_BUY_INTENT, response).apply()

    var buyResponse: Int
        get() = sharedPreferences.getInt(BUY, RESULT_DEFAULT)
        set(response) = sharedPreferences.edit().putInt(BUY, response).apply()

    var getPurchasesResponse: Int
        get() = sharedPreferences.getInt(GET_PURCHASES, RESULT_DEFAULT)
        internal set(response) = sharedPreferences.edit().putInt(GET_PURCHASES, response).apply()
    var getSkuDetailsResponse: Int
        get() = sharedPreferences.getInt(GET_SKU_DETAILS, RESULT_DEFAULT)
        internal set(response) = sharedPreferences.edit().putInt(GET_SKU_DETAILS, response).apply()

    var consumePurchaseResponse: Int
        get() = sharedPreferences.getInt(CONSUME_PURCHASE, RESULT_DEFAULT)
        internal set(response) = sharedPreferences.edit().putInt(CONSUME_PURCHASE, response).apply()

    var getBuyIntentToReplaceSkusResponse: Int
        get() = sharedPreferences.getInt(GET_BUY_INTENT_TO_REPLACE_SKUS, RESULT_DEFAULT)
        internal set(response) = sharedPreferences.edit().putInt(GET_BUY_INTENT_TO_REPLACE_SKUS, response).apply()

    val replaceResponse: Int
        get() = sharedPreferences.getInt(REPLACE, RESULT_DEFAULT)

    val usersResponse: String
        get() = sharedPreferences.getString(USERS, DEFAULT_USER)

    fun setReplaceReponse(response: Int) {
        sharedPreferences.edit().putInt(REPLACE, response).apply()
    }

    fun setUsersReponse(user: String) {
        sharedPreferences.edit().putString(USERS, user).apply()
    }

    companion object {
        const val PREF_NAME = "RegisterModel"
        const val CONFIG_FILE = BuildConfig.CONFIGURATION_FILE_NAME
        const val RESULT_DEFAULT = -1            // - no user override
        private const val IS_BILLING_SUPPORTED = "isBillingSupported"
        private const val GET_BUY_INTENT = "getBuyIntent"
        private const val BUY = "Buy"
        private const val GET_PURCHASES = "getPurchases"
        private const val GET_SKU_DETAILS = "getSkuDetails"
        private const val CONSUME_PURCHASE = "consumePurchase"
        private const val GET_BUY_INTENT_TO_REPLACE_SKUS = "getBuyIntentToReplaceSkus"
        private const val REPLACE = "Replace"
        private const val USERS = "Users"
        private const val DEFAULT_USER = "200nyttest1@nytimes.com"
    }
}
